lib/fs.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const metasync = require('metasync');
+
+const rmdirp = (dir, cb) => {
+  fs.stat(dir, (err, stats) => {
+    if (err) {
+      if (err.code === 'ENOENT') cb(null);
+      else cb(err);
+      return;
+    }
+
+    if (stats.isDirectory()) {
+      fs.readdir(dir, (err, files) => {
+        if (err) {
+          cb(err);
+          return;
+        }
+        files = files.map(file => path.join(dir, file));
+        metasync.each(files, rmdirp, err => {
+          if (err) cb(err);
+          else fs.rmdir(dir, cb);
+        });
+      });
+    } else {
+      fs.unlink(dir, cb);
+    }
+  });
+};
+
+const mkdir = (dir, cb) => {
+  fs.access(dir, fs.constants.F_OK, err => {
+    if (err && err.code === 'ENOENT') {
+      mkdir(path.dirname(dir), err => {
+        if (err) cb(err);
+        else fs.mkdir(dir, cb);
+      });
+    } else {
+      cb(err);
+    }
+  });
+};
+
+const mkdirp = (dir, cb) => {
+  const version = process.versions.node.split('.').map(el => parseInt(el));
+  if (version[0] < 10 || (version[0] === 10 && version[1] <= 11)) {
```

Comment:
Since we might have polyfills in other modules it will be better to extract it to a separate file. Though it should be done in another PR. WDYT?

Reply:
I don't see any reason for extracting it into a separate file.

Reply:
For example, at one point we were considering to have `Uint64` and `Int64` based on `BigInt`, though it is not relevant now. Also, some other modules might use it.

Reply:
And what's that have to do with `fs.mkdir()` polyfill?

Reply:
Since we will extract function for version compatibility checking, we might as well extract it to a separate file so it can be used by other modules. Though you have the point - this PR shouldn't rely on other PRs.

Reply:
I haven't proposed to extract the function for version compatibility checking, I've just said that the version check must be performed only once on this file `require()`, not every time the exported function runs since there is no way Node.js version changes between two calls.

Reply:
Yeah, but since we have version compatibility check in this file, I just thought that we might also implement a separate function for such checks. But, as I said, that's a discussion for a different PR.